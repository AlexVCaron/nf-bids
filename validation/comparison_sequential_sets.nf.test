nextflow_workflow {

    name "Plugin Comparison: Sequential Sets"
    script "./main.nf"
    workflow "main_workflow"

    test("qmri_vfa - Plugin output") {
        when {
            params {
                bids_dir = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples/qmri_vfa"
                libbids_sh = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/libBIDS.sh/libBIDS.sh"
                config = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/plugins/nf-bids/validation/configs/config_vfa.yaml"
            }
        }
        then {
            assert workflow.success
            assert workflow.out.bids_channel
            def results = workflow.out.bids_channel
            collect()
            def json = groovy.json.JsonOutput.toJson(results)
            def serialized = new groovy.json.JsonSlurper().parseText(json)
            assert snapshot(
                groovy.json.JsonOutput.toJson(serialized)
            ).match()
        }
    }

    test("qmri_irt1 - Plugin output") {
        when {
            params {
                bids_dir = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples/qmri_irt1"
                libbids_sh = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/libBIDS.sh/libBIDS.sh"
                config = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/plugins/nf-bids/validation/configs/config_irt1.yaml"
            }
        }
        then {
            assert workflow.success
            assert workflow.out.bids_channel
            def results = workflow.out.bids_channel
            collect()
            def json = groovy.json.JsonOutput.toJson(results)
            def serialized = new groovy.json.JsonSlurper().parseText(json)
            assert snapshot(
                groovy.json.JsonOutput.toJson(serialized)
            ).match()
        }
    }

    test("qmri_megre - Plugin output") {
        when {
            params {
                bids_dir = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples/qmri_megre"
                libbids_sh = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/libBIDS.sh/libBIDS.sh"
                config = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/plugins/nf-bids/validation/configs/config_megre.yaml"
            }
        }
        then {
            assert workflow.success
            assert workflow.out.bids_channel
            def results = workflow.out.bids_channel
            collect()
            def json = groovy.json.JsonOutput.toJson(results)
            def serialized = new groovy.json.JsonSlurper().parseText(json)
            assert snapshot(
                groovy.json.JsonOutput.toJson(serialized)
            ).match()
        }
    }

    test("qmri_mese - Plugin output") {
        when {
            params {
                bids_dir = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples/qmri_mese"
                libbids_sh = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/libBIDS.sh/libBIDS.sh"
                config = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/plugins/nf-bids/validation/configs/config_mese.yaml"
            }
        }
        then {
            assert workflow.success
            assert workflow.out.bids_channel
            def results = workflow.out.bids_channel
            collect()
            def json = groovy.json.JsonOutput.toJson(results)
            def serialized = new groovy.json.JsonSlurper().parseText(json)
            assert snapshot(
                groovy.json.JsonOutput.toJson(serialized)
            ).match()
        }
    }

    test("qmri_mp2rage - Plugin output") {
        when {
            params {
                bids_dir = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples/qmri_mp2rage"
                libbids_sh = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/libBIDS.sh/libBIDS.sh"
                config = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/plugins/nf-bids/validation/configs/config_mp2rage.yaml"
            }
        }
        then {
            assert workflow.success
            assert workflow.out.bids_channel
            def results = workflow.out.bids_channel
            collect()
            def json = groovy.json.JsonOutput.toJson(results)
            def serialized = new groovy.json.JsonSlurper().parseText(json)
            assert snapshot(
                groovy.json.JsonOutput.toJson(serialized)
            ).match()
        }
    }

    test("qmri_sa2rage - Plugin output") {
        when {
            params {
                bids_dir = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples/qmri_sa2rage"
                libbids_sh = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/libBIDS.sh/libBIDS.sh"
                config = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/plugins/nf-bids/validation/configs/config_sa2rage.yaml"
            }
        }
        then {
            assert workflow.success
            assert workflow.out.bids_channel
            def results = workflow.out.bids_channel
            collect()
            def json = groovy.json.JsonOutput.toJson(results)
            def serialized = new groovy.json.JsonSlurper().parseText(json)
            assert snapshot(
                groovy.json.JsonOutput.toJson(serialized)
            ).match()
        }
    }
}
