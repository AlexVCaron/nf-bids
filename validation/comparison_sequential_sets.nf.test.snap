{
    "qmri_megre - Plugin output": {
        "content": [
            "[[[\"sub-01\",\"NA\",\"NA\",\"NA\"],{\"data\":{\"MEGRE\":{\"json\":[\"qmri_megre/sub-01/anat/sub-01_echo-01_MEGRE.json\",\"qmri_megre/sub-01/anat/sub-01_echo-02_MEGRE.json\",\"qmri_megre/sub-01/anat/sub-01_echo-03_MEGRE.json\",\"qmri_megre/sub-01/anat/sub-01_echo-04_MEGRE.json\",\"qmri_megre/sub-01/anat/sub-01_echo-05_MEGRE.json\",\"qmri_megre/sub-01/anat/sub-01_echo-06_MEGRE.json\",\"qmri_megre/sub-01/anat/sub-01_echo-07_MEGRE.json\",\"qmri_megre/sub-01/anat/sub-01_echo-08_MEGRE.json\"],\"nii\":[\"qmri_megre/sub-01/anat/sub-01_echo-01_MEGRE.nii.gz\",\"qmri_megre/sub-01/anat/sub-01_echo-02_MEGRE.nii.gz\",\"qmri_megre/sub-01/anat/sub-01_echo-03_MEGRE.nii.gz\",\"qmri_megre/sub-01/anat/sub-01_echo-04_MEGRE.nii.gz\",\"qmri_megre/sub-01/anat/sub-01_echo-05_MEGRE.nii.gz\",\"qmri_megre/sub-01/anat/sub-01_echo-06_MEGRE.nii.gz\",\"qmri_megre/sub-01/anat/sub-01_echo-07_MEGRE.nii.gz\",\"qmri_megre/sub-01/anat/sub-01_echo-08_MEGRE.nii.gz\"]}},\"filePaths\":[\"qmri_megre/sub-01/anat/sub-01_echo-01_MEGRE.json\",\"qmri_megre/sub-01/anat/sub-01_echo-01_MEGRE.nii.gz\",\"qmri_megre/sub-01/anat/sub-01_echo-02_MEGRE.json\",\"qmri_megre/sub-01/anat/sub-01_echo-02_MEGRE.nii.gz\",\"qmri_megre/sub-01/anat/sub-01_echo-03_MEGRE.json\",\"qmri_megre/sub-01/anat/sub-01_echo-03_MEGRE.nii.gz\",\"qmri_megre/sub-01/anat/sub-01_echo-04_MEGRE.json\",\"qmri_megre/sub-01/anat/sub-01_echo-04_MEGRE.nii.gz\",\"qmri_megre/sub-01/anat/sub-01_echo-05_MEGRE.json\",\"qmri_megre/sub-01/anat/sub-01_echo-05_MEGRE.nii.gz\",\"qmri_megre/sub-01/anat/sub-01_echo-06_MEGRE.json\",\"qmri_megre/sub-01/anat/sub-01_echo-06_MEGRE.nii.gz\",\"qmri_megre/sub-01/anat/sub-01_echo-07_MEGRE.json\",\"qmri_megre/sub-01/anat/sub-01_echo-07_MEGRE.nii.gz\",\"qmri_megre/sub-01/anat/sub-01_echo-08_MEGRE.json\",\"qmri_megre/sub-01/anat/sub-01_echo-08_MEGRE.nii.gz\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-01\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"NA\"}]]"
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.10.0"
        },
        "timestamp": "2025-10-28T23:43:40.329782865"
    },
    "qmri_vfa - Plugin output": {
        "content": [
            "[[[\"sub-01\",\"NA\",\"NA\",\"NA\"],{\"data\":{\"TB1AFI\":{\"tr1\":{\"json\":\"qmri_vfa/sub-01/fmap/sub-01_acq-tr1_TB1AFI.json\",\"nii\":\"qmri_vfa/sub-01/fmap/sub-01_acq-tr1_TB1AFI.nii.gz\"},\"tr2\":{\"json\":\"qmri_vfa/sub-01/fmap/sub-01_acq-tr2_TB1AFI.json\",\"nii\":\"qmri_vfa/sub-01/fmap/sub-01_acq-tr2_TB1AFI.nii.gz\"}},\"VFA\":{\"json\":[\"qmri_vfa/sub-01/anat/sub-01_flip-1_VFA.json\",\"qmri_vfa/sub-01/anat/sub-01_flip-2_VFA.json\"],\"nii\":[\"qmri_vfa/sub-01/anat/sub-01_flip-1_VFA.nii.gz\",\"qmri_vfa/sub-01/anat/sub-01_flip-2_VFA.nii.gz\"]}},\"filePaths\":[\"qmri_vfa/sub-01/fmap/sub-01_acq-tr1_TB1AFI.json\",\"qmri_vfa/sub-01/fmap/sub-01_acq-tr1_TB1AFI.nii.gz\",\"qmri_vfa/sub-01/fmap/sub-01_acq-tr2_TB1AFI.json\",\"qmri_vfa/sub-01/fmap/sub-01_acq-tr2_TB1AFI.nii.gz\",\"qmri_vfa/sub-01/anat/sub-01_flip-1_VFA.json\",\"qmri_vfa/sub-01/anat/sub-01_flip-1_VFA.nii.gz\",\"qmri_vfa/sub-01/anat/sub-01_flip-2_VFA.json\",\"qmri_vfa/sub-01/anat/sub-01_flip-2_VFA.nii.gz\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-01\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"NA\"}]]"
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.10.0"
        },
        "timestamp": "2025-10-28T23:43:29.825910464"
    },
    "qmri_mese - Plugin output": {
        "content": [
            "[[[\"sub-01\",\"NA\",\"NA\",\"NA\"],{\"data\":{\"MESE\":{\"json\":[\"qmri_mese/sub-01/anat/sub-01_echo-01_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-02_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-03_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-04_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-05_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-06_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-07_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-08_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-09_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-10_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-11_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-12_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-13_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-14_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-15_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-16_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-17_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-18_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-19_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-20_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-21_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-22_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-23_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-24_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-25_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-26_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-27_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-28_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-29_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-30_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-31_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-32_MESE.json\"],\"nii\":[\"qmri_mese/sub-01/anat/sub-01_echo-01_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-02_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-03_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-04_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-05_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-06_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-07_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-08_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-09_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-10_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-11_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-12_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-13_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-14_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-15_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-16_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-17_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-18_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-19_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-20_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-21_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-22_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-23_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-24_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-25_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-26_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-27_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-28_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-29_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-30_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-31_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-32_MESE.nii.gz\"]}},\"filePaths\":[\"qmri_mese/sub-01/anat/sub-01_echo-01_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-01_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-02_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-02_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-03_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-03_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-04_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-04_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-05_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-05_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-06_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-06_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-07_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-07_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-08_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-08_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-09_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-09_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-10_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-10_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-11_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-11_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-12_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-12_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-13_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-13_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-14_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-14_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-15_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-15_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-16_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-16_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-17_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-17_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-18_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-18_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-19_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-19_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-20_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-20_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-21_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-21_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-22_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-22_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-23_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-23_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-24_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-24_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-25_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-25_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-26_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-26_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-27_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-27_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-28_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-28_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-29_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-29_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-30_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-30_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-31_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-31_MESE.nii.gz\",\"qmri_mese/sub-01/anat/sub-01_echo-32_MESE.json\",\"qmri_mese/sub-01/anat/sub-01_echo-32_MESE.nii.gz\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-01\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"NA\"}]]"
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.10.0"
        },
        "timestamp": "2025-10-28T23:43:46.035431"
    },
    "qmri_sa2rage - Plugin output": {
        "content": [
            "[[[\"sub-01\",\"NA\",\"NA\",\"NA\"],{\"data\":{\"TB1SRGE\":{\"json\":[[\"qmri_sa2rage/sub-01/fmap/sub-01_flip-1_inv-1_TB1SRGE.json\"],[\"qmri_sa2rage/sub-01/fmap/sub-01_flip-2_inv-2_TB1SRGE.json\"]],\"nii\":[[\"qmri_sa2rage/sub-01/fmap/sub-01_flip-1_inv-1_TB1SRGE.nii.gz\"],[\"qmri_sa2rage/sub-01/fmap/sub-01_flip-2_inv-2_TB1SRGE.nii.gz\"]]}},\"filePaths\":[\"qmri_sa2rage/sub-01/fmap/sub-01_flip-1_inv-1_TB1SRGE.json\",\"qmri_sa2rage/sub-01/fmap/sub-01_flip-1_inv-1_TB1SRGE.nii.gz\",\"qmri_sa2rage/sub-01/fmap/sub-01_flip-2_inv-2_TB1SRGE.json\",\"qmri_sa2rage/sub-01/fmap/sub-01_flip-2_inv-2_TB1SRGE.nii.gz\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-01\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"NA\"}]]"
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.10.0"
        },
        "timestamp": "2025-10-29T11:06:52.413022501"
    },
    "qmri_irt1 - Plugin output": {
        "content": [
            "[[[\"sub-01\",\"NA\",\"NA\",\"NA\"],{\"data\":{\"IRT1\":{\"json\":[\"qmri_irt1/sub-01/anat/sub-01_inv-01_IRT1.json\",\"qmri_irt1/sub-01/anat/sub-01_inv-02_IRT1.json\",\"qmri_irt1/sub-01/anat/sub-01_inv-03_IRT1.json\",\"qmri_irt1/sub-01/anat/sub-01_inv-04_IRT1.json\"],\"nii\":[\"qmri_irt1/sub-01/anat/sub-01_inv-01_IRT1.nii.gz\",\"qmri_irt1/sub-01/anat/sub-01_inv-02_IRT1.nii.gz\",\"qmri_irt1/sub-01/anat/sub-01_inv-03_IRT1.nii.gz\",\"qmri_irt1/sub-01/anat/sub-01_inv-04_IRT1.nii.gz\"]}},\"filePaths\":[\"qmri_irt1/sub-01/anat/sub-01_inv-01_IRT1.json\",\"qmri_irt1/sub-01/anat/sub-01_inv-01_IRT1.nii.gz\",\"qmri_irt1/sub-01/anat/sub-01_inv-02_IRT1.json\",\"qmri_irt1/sub-01/anat/sub-01_inv-02_IRT1.nii.gz\",\"qmri_irt1/sub-01/anat/sub-01_inv-03_IRT1.json\",\"qmri_irt1/sub-01/anat/sub-01_inv-03_IRT1.nii.gz\",\"qmri_irt1/sub-01/anat/sub-01_inv-04_IRT1.json\",\"qmri_irt1/sub-01/anat/sub-01_inv-04_IRT1.nii.gz\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-01\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"NA\"}]]"
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.10.0"
        },
        "timestamp": "2025-10-28T23:43:35.139992985"
    },
    "qmri_mp2rage - Plugin output": {
        "content": [
            "[[[\"sub-1\",\"NA\",\"NA\",\"NA\"],{\"data\":{\"MP2RAGE\":{\"nii\":[{\"mag\":\"qmri_mp2rage/sub-1/anat/sub-1_inv-1_part-mag_MP2RAGE.nii\",\"phase\":\"qmri_mp2rage/sub-1/anat/sub-1_inv-1_part-phase_MP2RAGE.nii\"},{\"mag\":\"qmri_mp2rage/sub-1/anat/sub-1_inv-2_part-mag_MP2RAGE.nii\",\"phase\":\"qmri_mp2rage/sub-1/anat/sub-1_inv-2_part-phase_MP2RAGE.nii\"}],\"json\":[\"qmri_mp2rage/sub-1/anat/sub-1_inv-1_MP2RAGE.json\",\"qmri_mp2rage/sub-1/anat/sub-1_inv-2_MP2RAGE.json\"]},\"UNIT1\":{\"json\":\"qmri_mp2rage/derivatives/pymp2rage/sub-1/anat/sub-1_UNIT1.json\",\"nii\":\"qmri_mp2rage/sub-1/anat/sub-1_UNIT1.nii\"}},\"filePaths\":[\"qmri_mp2rage/sub-1/anat/sub-1_inv-1_MP2RAGE.json\",\"qmri_mp2rage/sub-1/anat/sub-1_inv-1_part-mag_MP2RAGE.nii\",\"qmri_mp2rage/sub-1/anat/sub-1_inv-1_part-phase_MP2RAGE.nii\",\"qmri_mp2rage/sub-1/anat/sub-1_inv-2_MP2RAGE.json\",\"qmri_mp2rage/sub-1/anat/sub-1_inv-2_part-mag_MP2RAGE.nii\",\"qmri_mp2rage/sub-1/anat/sub-1_inv-2_part-phase_MP2RAGE.nii\",\"qmri_mp2rage/derivatives/pymp2rage/sub-1/anat/sub-1_UNIT1.json\",\"qmri_mp2rage/derivatives/pymp2rage/sub-1/anat/sub-1_UNIT1.nii\",\"qmri_mp2rage/sub-1/anat/sub-1_UNIT1.nii\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-1\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"NA\"}]]"
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.10.0"
        },
        "timestamp": "2025-10-29T14:02:11.779946157"
    }
}